app/src/main/java/com/example/parkingsystem/mvp/presenter/ParkingPresenter.java

```diff
@@ -13,13 +13,19 @@ public ParkingPresenter(ParkingContract.Model model, ParkingContract.View view)
         this.view = view;
     }
 
+    @Override
+    public void inflateDialog(ListenerDialogFragment listenerDialogFragment) {
+        view.showDialog(listenerDialogFragment);
+    }
+
+    @Override
     public void onSetParkingPlacesButtonPressed(int spaces) {
         model.setSpaces(spaces);
         view.showPopUp(model.getSpaces());
     }
 
     @Override
-    public void inflateDialog(ListenerDialogFragment listenerDialogFragment) {
-        view.showDialog(listenerDialogFragment);
+    public void onReservationButtonClicked() {
+        view.startNewActivity();
```

Comment:
you should follow the same naming conventions for simplicity. The name startNewActivity is too generic you can change it by openReservationScreen()

Reply:
Done